### ISSUE TITLE
Support boolean datatype

### ISSUE THREAD

**Issue by loichuder** [usnistgov/h5wasm#22]
Boolean datatype is treated as `H5T_ENUM` (when saved though h5py):
```
      DATATYPE  H5T_ENUM {
         H5T_STD_I8LE;
         "FALSE"            0;
         "TRUE"             1;
      }
```

With the implementation of 0.4.1, h5wasm returns the metadata and dtype of the base type `integer` which do not allow to distinguish between integer and boolean datasets.

Ideally, `H5T_ENUM` should be marked as such. The base type could be stored in a sub-field `enum_type` as for compound and array dtypes.

**Comment by bmaranville** [comment]
Sure - this is no problem, if it is important for your use case.  I think #21 might have to be completed and merged first, to avoid confusion.

**Comment by loichuder** [comment]
Yes ! But I think your commit just before 4682f99e5400d531e31ea21e3de9e3c4edd1c50f was even better with the base type part of `enum_type`:

```js
{
    "signed": true,
    "cset": -1,
    "enum_type": {
        "nmembers": 2,
        "members": [
            "FALSE",
            "TRUE"
        ],
        "type": 0,
    },
    "vlen": false,
    "littleEndian": true,
    "size": 1,
    "shape": [
        3
    ],
    "total_size": 3
}
```

The only missing field would be `type` that should be set to `8` (`H5T_ENUM`) via
```c
attr.set("type", (int)dtype_class);
```

**Comment by bmaranville** [comment]
I think this feature is complete as of e5c4d6c24642e87ea53b69eccabc67f39c1e89a1 , including your suggestion of leaving the ENUM type (8) in the metadata and putting the base type in ```enum_type.type```
